python/ribasim/ribasim/config.py

```diff
@@ -0,0 +1,117 @@
+    maxiters: Optional[int] = Field(1000000000, description="maxiters")
+    autodiff: Optional[bool] = Field(True, description="autodiff")
+    adaptive: Optional[bool] = Field(True, description="adaptive")
+    algorithm: Optional[str] = Field("QNDF", description="algorithm")
+    abstol: Optional[float] = Field(1e-06, description="abstol")
+    dt: Optional[float] = Field(0, description="dt")
+    sparse: Optional[bool] = Field(True, description="sparse")
+
+
+class FlowBoundary(BaseModel):
+    time: Optional[str] = Field(None, description="time")
+    static: Optional[str] = Field(None, description="static")
+
+
+class PidControl(BaseModel):
+    time: Optional[str] = Field(None, description="time")
+    static: Optional[str] = Field(None, description="static")
+
+
+class FractionalFlow(BaseModel):
+    static: Optional[str] = Field(None, description="static")
+
+
+class ManningResistance(BaseModel):
+    static: Optional[str] = Field(None, description="static")
+
+
+class TabulatedRatingCurve(BaseModel):
+    time: Optional[str] = Field(None, description="time")
+    static: Optional[str] = Field(None, description="static")
+
+
+class Logging(BaseModel):
+    timing: Optional[bool] = Field(False, description="timing")
+    verbosity: Optional[str] = Field("info", description="verbosity")
+
+
+class Outlet(BaseModel):
+    static: Optional[str] = Field(None, description="static")
+
+
+class Terminal(BaseModel):
+    static: Optional[str] = Field(None, description="static")
+
+
+class Basin(BaseModel):
+    profile: Optional[str] = Field(None, description="profile")
+    static: Optional[str] = Field(None, description="static")
+    forcing: Optional[str] = Field(None, description="forcing")
+    state: Optional[str] = Field(None, description="state")
+
+
+class LinearResistance(BaseModel):
+    static: Optional[str] = Field(None, description="static")
+
+
+class Config(BaseModel):
+    output: Optional[Output] = None
+    starttime: datetime = Field(..., description="starttime")
+    update_timestep: Optional[float] = Field(86400, description="update_timestep")
+    input_dir: Optional[str] = Field(".", description="input_dir")
+    output_dir: Optional[str] = Field(".", description="output_dir")
+    level_boundary: Optional[LevelBoundary] = None
+    pump: Optional[Pump] = None
+    discrete_control: Optional[DiscreteControl] = None
+    solver: Optional[Solver] = None
+    flow_boundary: Optional[FlowBoundary] = None
+    pid_control: Optional[PidControl] = None
+    fractional_flow: Optional[FractionalFlow] = None
+    relative_dir: Optional[str] = Field(".", description="relative_dir")
+    endtime: datetime = Field(..., description="endtime")
+    manning_resistance: Optional[ManningResistance] = None
+    tabulated_rating_curve: Optional[TabulatedRatingCurve] = None
+    logging: Optional[Logging] = None
+    outlet: Optional[Outlet] = None
+    geopackage: str = Field(..., description="geopackage")
```

Comment:
What is `...`?

And what is the use for the description, if it is the same as the field name? And why is it sometimes there and sometimes not?

Reply:
It's how Pydantic sets a Field to be required. I can probably remove the description by changing a command line flag, but it's not used anyway.

Reply:
Ah ok. If it's not used and you can find a flag to remove it, I'd prefer that.